api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/TxMlQueries.java

```diff
@@ -363,14 +410,55 @@ public static String getPatientsWithVisitCardAndWithObs(
             + "  INNER JOIN encounter e ON pa.patient_id=e.patient_id "
             + "  INNER JOIN obs o ON pa.patient_id=o.person_id "
             + "  INNER JOIN ("
-            + "    SELECT p.patient_id,MAX(e.encounter_datetime) return_date FROM patient p "
-            + "      INNER JOIN encounter e ON e.patient_id = p.patient_id AND e.encounter_datetime <=:endDate AND e.location_id=:location "
-            + "      INNER JOIN obs o ON o.encounter_id = e.encounter_id  AND o.location_id=:location  "
-            + "    WHERE p.voided = 0 AND e.voided = 0 AND o.voided=0 "
-            + "      AND e.encounter_type IN (%d,%d,%d) "
-            + "      AND o.concept_id IN (%d, %d) "
-            + "      AND e.location_id =:location "
-            + "    GROUP BY p.patient_id)lp ON pa.patient_id=lp.patient_id "
+            + "                SELECT patient_id, MAX(return_date) AS return_date "
+            + "                FROM (SELECT p.patient_id, "
+            + "                             (SELECT o.value_datetime AS return_date "
+            + "                              FROM encounter e "
+            + "                                       INNER JOIN obs o "
+            + "                                                  ON e.encounter_id = o.encounter_id "
+            + "                              WHERE p.patient_id = e.patient_id "
+            + "                                AND e.voided = 0 "
+            + "                                AND o.voided = 0 "
+            + "                                AND e.encounter_type = %d "
+            + "                                AND e.location_id = :location "
+            + "                                AND o.concept_id = %d "
+            + "                                AND e.encounter_datetime <= :endDate "
+            + "                              ORDER BY e.encounter_datetime DESC "
+            + "                              LIMIT 1) AS return_date "
+            + "                      FROM patient p "
+            + "                      UNION "
+            + "                      SELECT p.patient_id, "
```

Comment:
the spec says the most recent scheduled date, on my understanding it looks like this query is picking most recent encounter